openedx/features/personalized_learner_schedules/call_to_action.py

```diff
@@ -25,14 +26,18 @@ def get_ctas(self, xblock, category):
         Look at CallToActionService docstring to see what will be returned.
         """
         ctas = []
+        request = get_current_request()
+
+        course_key = xblock.scope_ids.usage_id.context_key
+        missed_deadlines, missed_gated_content = dates_banner_should_display(course_key, request.user)
+        if not missed_deadlines or (missed_deadlines and missed_gated_content):
```

Comment:
This is more defensive and so, fine. But `missed_gated_content` should never be true without `missed_deadlines` also being true.

Reply:
I didn't even realize that. I was just trying to take the scenarios from dates_banner.html and replicate here. That's a good point though. I can update to just `if not missed_deadlines or missed_gated_content:`

Reply:
Oh yeah, even if that weren't true (and I believe it is), you can reduce to `not missed_deadlines or missed_gated_content` since the first part already means that in the second part, `missed_deadlines` will be true.